clang/lib/Driver/ToolChains/Flang.cpp

```diff
@@ -492,6 +493,18 @@ void Flang::addOffloadOptions(Compilation &C, const InputInfoList &Inputs,
     if (Args.hasArg(options::OPT_nogpulib))
       CmdArgs.push_back("-nogpulib");
   }
+
+  // For all the host OpenMP offloading compile jobs we need to pass the targets
+  // information using -fopenmp-targets= option.
+  if (JA.isHostOffloading(Action::OFK_OpenMP)) {
+    SmallString<128> Targets("-fopenmp-targets=");
```

Comment:
Or, even better, `StringLiteral`.

Reply:
> Would it be better to leave it unchanged, let both implementations diverge or apply these suggestions to clang too? I think it makes sense to do them, just trying to avoid divergence on duplicated code.

Unfortunately, it will diverge over time. Bugs will be fixed for `Clang.cpp` but forget to apply it to `Flang.cpp` as well. For now, I would grudgingly accept the code as-is if this was done for the rest of the file, but hope that we can avoid all this code duplication in the future. 

Did you consider to just extract the lines from `Clang.cpp` into another function to some location this is accessible to both?

Reply:
> For now, I would grudgingly accept the code as-is if this was done for the rest of the file, but hope that we can avoid all this code duplication in the future.
> 
> Did you consider to just extract the lines from `Clang.cpp` into another function to some location this is accessible to both?

I agree that we should try to avoid duplicating code like this as much as possible. In this case, I just followed what's been done for other similar options and just copied it from clang.

After a quick look, I don't see an obvious place to add this shared code, but it looks like one way to get there would be to create a `Driver` or similar class deriving from `Tool` and make that the shared parent for the `Clang` and `Flang` classes, putting in it common option processing code used by both. Something like this would probably be best done on its own patch after some discussion, though. In any case, I think @banach-space is the expert on this area, so maybe he knows of any plans already in place to improve this situation or can give us some pointers on how to better deal with this in the short term.

Reply:
This looks to be taken verbatim from Clang:
* https://github.com/llvm/llvm-project/blob/2ba3fe7356f065757a2279f65e4ef5c8f1476293/clang/lib/Driver/ToolChains/Clang.cpp#L7768-L7778

It should be extracted to a dedicated hook and moved to https://github.com/llvm/llvm-project/blob/main/clang/lib/Driver/ToolChains/CommonArgs.cpp.  That's the best we have today. 

>  I think @banach-space is the expert on this area, so maybe he knows of any plans already in place to improve this situation or can give us some pointers on how to better deal with this in the short term.

These days I only have very limited bandwidth to _review_ driver patches. Please consider every contribution that you make as an opportunity to improve code re-use between Clang and Flang.

Reply:
Thank you for the pointer, @banach-space. I moved this into a new function in CommonArgs, so there's no longer code duplication added for this. Addressed @Meinersbur comments as well.